### PR TITLE
Add link to all bookings in room details

### PR DESCRIPTION
**Pull request by meluru**
Screenshot_2019-06-28 Room Booking · Indico

### CODE REVIEW

File: indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx [hunk 1]
```diff
@@ -106,6 +113,14 @@ export default connect(
     attributes: roomsSelectors.getAttributes(state, {roomId}),
   }),
   dispatch => ({
+    gotoAllBookings(roomName) {
+      const params = {
+        text: roomName,
+      };
+      dispatch(globalActions.resetPageState('calendar'));
+      dispatch(filtersActions.setFilters('calendar', {text: roomName}, false));
```
Comment (mic4ael) [inline indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx]: replace the `{text:` with `params`?

File: indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx [hunk 2]
```diff
@@ -171,6 +186,16 @@ function RoomDetails({bookRoom, room, availability, attributes}) {
               trigger={<Icon name="info circle" className="legend-info-icon" />}
               content={<TimelineLegend labels={legendLabels} compact />}
             />
+            <Button
```
Comment (ThiefMaster) [inline indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx]: could we render this button as an `<a />` with a suitable `href`, so people can middle-click it and open a new tab with the results?

File: indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx [hunk 3]
```diff
@@ -106,6 +113,14 @@ export default connect(
     attributes: roomsSelectors.getAttributes(state, {roomId}),
   }),
   dispatch => ({
+    gotoAllBookings(roomName) {
+      const params = {
+        text: roomName,
+      };
+      dispatch(globalActions.resetPageState('calendar'));
+      dispatch(filtersActions.setFilters('calendar', {text: roomName}, false));
+      dispatch(pushRoute(`/calendar?${qs.stringify(params)}`));
```
Comment (mvidalgarcia) [inline indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx]: What about setting the month view mode? Redirecting to the day view displays less information than in the room details.

File: indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx (hunk 2 above)
Comment (meluru) [reply indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx]: that's a good idea, I'll give it a try

File: indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx (hunk 3 above)
Comment (meluru) [reply indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx]: Yeah, I'll do it if there's a way. Redirecting to calendar page with mode=months in the url doesn't work

File: indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx (hunk 1 above)
Comment (meluru) [reply indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx]: I'll change that part to id anyways :)

File: indico/modules/rb/operations/rooms.py [hunk 4]
```diff
@@ -152,7 +152,10 @@ def search_for_rooms(filters, allow_admin=False, availability=None):
     query = query.filter_by(**criteria)
     if 'text' in filters:
         text = ' '.join(filters['text'].strip().split())
-        query = query.filter(_make_room_text_filter(text))
+        if text.startswith('#') and text[1:].isdigit():
+            query = query.filter(Room.id == text[1:])
```
Comment (ThiefMaster) [inline indico/modules/rb/operations/rooms.py]: I think `int(text[1:])` would be cleaner here

File: indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx [hunk 5]
```diff
@@ -171,6 +192,23 @@ function RoomDetails({bookRoom, room, availability, attributes}) {
...
+            <Button
+              as="a"
+              href={bookingsPath}
+              basic
+              size="tiny"
+              compact
+              color="blue"
+              styleName="all-bookings"
+              onClick={evt => {
+                if (evt.button !== 1) {
```
Comment (ThiefMaster) [inline indico/modules/rb/client/js/common/rooms/RoomDetailsModal.jsx]: shouldn't this be `=== 0`? If we somehow end up getting in here for a right click (button 2) we don't want to do anything...

**Pull request merged**